doc/architecture/decisions/0012-handling-feature-flags-with-flipt.md

```diff
@@ -0,0 +1,59 @@
+# 12. Handling feature flags with Flipt
+
+Date: 2024-03-25
+
+## Status
+
+Accepted
+
+## Context
+
+As the service gets used by more people, and communicating new features / changes with
+users becomes more essential, it's important that sometimes we hide certain features
+behind a feature flag to ensure we can roll out features in a controlled manner, whilst
+not slowing down the development process.
+
+Previously we've handled feature flags in an adhoc way, by use of environment variables,
+but this causes a couple of issues, most importantly, to roll back or introduce a feature
+we need to start a new deploy, which can take time, as we need to go through the whole
+deployment process, from PR, through to E2E tests and deployment to various different
+environments. We also need to have a lot of code that handles fetching these variables,
+adding conditionals etc, which gives us more code to maintain.
+
+## Decision
+
+The integrations team have recently deployed an instance of [Flipt](https://www.flipt.io/),
+which is a simple service that allows us to manage feature flags and turn them on and off
+at will via a UI, without having to redeploy the service.
+
+There are instances for each environment, Dev, Prod and Prepod, meaning we can have
+different values for feature flags in each environment.
```

Comment:
will we be able to expand this to our 4th environment `test` which we may be repurposing on the UI side as a `qa`?

Reply:
Possibly, but we'd have to deploy another instance of the Flipt infrastructure